### ISSUE TITLE
Can't disable SC2120 for one function

### ISSUE THREAD

**Issue by GoldsteinE** [koalaman/shellcheck#1770]
- Rule Id (if any, e.g. SC1000): SC2120
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

# shellcheck disable=SC2120
f() {
        echo "${1:-foo}"
}

g() {
        echo "$1"
}

h() {
        f
        g
}
```

#### Here's what shellcheck currently says:
Nothing
#### Here's what I wanted or expected to see:
Report no error for `f` invocation but report an error for `g` invocation.

**Comment by Gandalf-** [comment]
Shellcheck is applying your first directive to the first file - it can't tell the difference here.

If you put another statement before your function declaration, this works. It's a work around, but maybe acceptable?

For instance
```shell
#!/bin/sh

: # the colon command has no effect here

# shellcheck disable=SC2120
f() {
  echo "${1:-foo}"
}

g() {
  echo "$1"
}

h() {
  f
  g
}
```
```
In 1770.sh line 10:
g() {
^-- SC2120: g references arguments, but none are ever passed.


In 1770.sh line 16:
  g
  ^-- SC2119: Use g "$@" if function's $1 should mean script's $1
```

**Comment by GoldsteinE** [comment]
It would be cool if shellcheck displayed a warning like "shellcheck can't tell if this directive applied to function or whole file" in these cases. Or applied directive to function, if there is no blank line below it and there is above.
Workaround is acceptable, thanks
